Makefile

```diff
@@ -113,6 +113,47 @@ distclean:
 
 check: test
 
+# Remove files generated by running coverage, put the un-intstrumented lib back in place 
+clean_coverage:
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	-rm -rf gcovr
+	-rm -rf out/Release/.coverage
+	-rm -rf .cov_tmp coverage
+	-rm -f out/Release/obj.target/node/src/*.gcda
+	-rm -f out/Release/obj.target/node/src/tracing/*.gcda
+	-rm -f out/Release/obj.target/node/src/*.gcno
+	-rm -f out/Release/obj.target/node/src/tracing*.gcno
+
+# Build and test with code coverage reporting.  Leave the lib directory instrumented for any additional runs the user may want to make
+# For C++ coverage reporting, this needs to be run in conjunction with configure --coverage.  html coverage reports will be 
+# created under ./coverage
+coverage: all
+	mkdir node_modules
+	$(NODE) ./deps/npm install istanbul-merge
+	$(NODE) ./deps/npm install nyc
+	git clone --depth=10 --single-branch git://github.com/gcovr/gcovr.git
+	git clone https://github.com/nodejs/testing.git
+	(cd gcovr && patch -p1 < "$(CURDIR)/testing/coverage/gcovr-patches.diff")
```

Comment:
Yes :) ... and I'm happy to take that on.  I took a quick look through gcovr, which essentially only revealed that I'd need to invest more time to take a proper look.